# Análisis Exploratorio de Datos (EDA)
Dataset: vehicles_us.csv

En este notebook se realiza un análisis exploratorio básico del conjunto de datos de anuncios de venta de coches en Estados Unidos.

In [47]:
import pandas as pd
import plotly.express as px

In [48]:
car_data = pd.read_csv("c:/programacion/triple10/proyecto_sprint7_vehicles/vehicles_us.csv")
car_data.head()

,price,model_year,model,condition,cylinders,fuel,odometer,transmission,type,paint_color,is_4wd,date_posted,days_listed
0,9400,2011.0,bmw x5,good,6.0,gas,145000.0,automatic,SUV,NaN,1.0,2018-06-23,19
1,25500,NaN,ford f-150,good,6.0,gas,88705.0,automatic,pickup,white,1.0,2018-10-19,50
2,5500,2013.0,hyundai sonata,like new,4.0,gas,110000.0,automatic,sedan,red,NaN,2019-02-07,79
3,1500,2003.0,ford f-150,fair,8.0,gas,NaN,automatic,pickup,NaN,NaN,2019-03-22,9
4,14900,2017.0,chrysler 200,excellent,4.0,gas,80903.0,automatic,sedan,black,NaN,2019-04-02,28


In [49]:
car_data.info()

<class 'pandas.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   price         51525 non-null  int64  
 1   model_year    47906 non-null  float64
 2   model         51525 non-null  str    
 3   condition     51525 non-null  str    
 4   cylinders     46265 non-null  float64
 5   fuel          51525 non-null  str    
 6   odometer      43633 non-null  float64
 7   transmission  51525 non-null  str    
 8   type          51525 non-null  str    
 9   paint_color   42258 non-null  str    
 10  is_4wd        25572 non-null  float64
 11  date_posted   51525 non-null  str    
 12  days_listed   51525 non-null  int64  
dtypes: float64(4), int64(2), str(7)
memory usage: 5.1 MB


## Limpieza mínima de datos

Se eliminan filas con valores faltantes en columnas necesarias para el análisis gráfico:
- odometer
- model_year


In [50]:
# Copiamos el dataset original para no modificarlo directamente
clean_data = car_data.copy()

# 1. Eliminamos filas sin kilometraje (necesario para histograma y scatter)
clean_data = clean_data.dropna(subset=["odometer"])

# 2. Eliminamos filas sin año del modelo
clean_data = clean_data.dropna(subset=["model_year"])

# 3. Convertimos model_year a entero
clean_data["model_year"] = clean_data["model_year"].astype(int)

# 4. Convertimos is_4wd a booleano (1 = True, NaN = False)
clean_data["is_4wd"] = clean_data["is_4wd"].fillna(0).astype(bool)

# Verificamos resultado
clean_data.info()

<class 'pandas.DataFrame'>
Index: 40563 entries, 0 to 51523
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   price         40563 non-null  int64  
 1   model_year    40563 non-null  int64  
 2   model         40563 non-null  str    
 3   condition     40563 non-null  str    
 4   cylinders     36419 non-null  float64
 5   fuel          40563 non-null  str    
 6   odometer      40563 non-null  float64
 7   transmission  40563 non-null  str    
 8   type          40563 non-null  str    
 9   paint_color   33306 non-null  str    
 10  is_4wd        40563 non-null  bool   
 11  date_posted   40563 non-null  str    
 12  days_listed   40563 non-null  int64  
dtypes: bool(1), float64(2), int64(3), str(7)
memory usage: 4.1 MB


In [51]:
fig = px.histogram(
    clean_data,
    x="odometer",
    title="Distribución del kilometraje",
)

fig.show()

In [52]:
fig = px.scatter(
    clean_data,
    x="odometer",
    y="price",
    opacity=0.5,
    title="Relación entre kilometraje y precio"
)

fig.show()

In [53]:
print("Filas originales:", car_data.shape[0])
print("Filas después de limpieza:", clean_data.shape[0])
print("Filas eliminadas:", car_data.shape[0] - clean_data.shape[0])

Filas originales: 51525
Filas después de limpieza: 40563
Filas eliminadas: 10962


## Conclusiones del Análisis Exploratorio de Datos (EDA)

### 1. Limpieza de datos

El conjunto de datos original contenía 51,525 registros.  
Después de eliminar filas con valores faltantes en las columnas `odometer` y `model_year`, quedaron 40,563 registros válidos para el análisis.

Se eliminaron 10,962 registros incompletos, lo que representa aproximadamente el 21% del dataset original.

La limpieza fue mínima y justificada, ya que estas variables son necesarias para construir visualizaciones confiables.

---

### 2. Distribución del kilometraje (Histograma)

El histograma muestra una distribución claramente sesgada hacia la derecha:

- La mayoría de los vehículos se concentran entre 0 y 200,000 millas.
- Existe una cola larga que se extiende hasta aproximadamente 1,000,000 de millas.
- Se observan valores extremos (outliers) con kilometraje muy elevado.

Esto indica que la mayor parte de los autos en venta tienen un uso moderado, mientras que un pequeño grupo presenta un desgaste significativamente mayor.

---

### 3. Relación entre kilometraje y precio (Gráfico de dispersión)

El gráfico de dispersión muestra una relación negativa clara entre el kilometraje y el precio:

- A medida que aumenta el kilometraje, el precio tiende a disminuir.
- Los vehículos con menor kilometraje presentan mayor variabilidad en precios.
- Se observan algunos outliers con precios altos y kilometrajes elevados, así como vehículos extremadamente baratos.

Esta tendencia es consistente con el comportamiento esperado del mercado automotriz: el desgaste acumulado impacta directamente el valor del vehículo.

---

### 4. Observaciones generales

- La relación entre precio y kilometraje es visualmente significativa.
- Existen valores extremos tanto en kilometraje como en precio que podrían influir en análisis más avanzados.
- La limpieza aplicada permitió obtener visualizaciones más coherentes y representativas del comportamiento real de los datos.

En conclusión, el dataset muestra patrones consistentes con el mercado de autos usados y está en condiciones adecuadas para integrarse en la aplicación web desarrollada con Streamlit.